server/service/activities.go

```diff
@@ -99,3 +99,40 @@ func (svc *Service) ListHostUpcomingActivities(ctx context.Context, hostID uint,
 
 	return svc.ds.ListHostUpcomingActivities(ctx, hostID, opt)
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// List host past activities
+////////////////////////////////////////////////////////////////////////////////
+
+type listHostPastActivitiesRequest struct {
+	HostID      uint              `url:"id"`
+	ListOptions fleet.ListOptions `url:"list_options"`
+}
+
+func listHostPastActivitiesEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
+	req := request.(*listHostPastActivitiesRequest)
+	acts, meta, err := svc.ListHostPastActivities(ctx, req.HostID, req.ListOptions)
+	if err != nil {
+		return listActivitiesResponse{Err: err}, nil
+	}
+
+	return &listActivitiesResponse{Meta: meta, Activities: acts}, nil
+}
+
+func (svc *Service) ListHostPastActivities(ctx context.Context, hostID uint, opt fleet.ListOptions) ([]*fleet.Activity, *fleet.PaginationMetadata, error) {
+	// First ensure the user has access to list hosts, then check the specific
+	// host once team_id is loaded.
+	if err := svc.authz.Authorize(ctx, &fleet.Host{}, fleet.ActionList); err != nil {
+		return nil, nil, err
+	}
+	host, err := svc.ds.HostLite(ctx, hostID)
+	if err != nil {
+		return nil, nil, ctxerr.Wrap(ctx, err, "get host")
+	}
+	// Authorize again with team loaded now that we have team_id
+	if err := svc.authz.Authorize(ctx, host, fleet.ActionRead); err != nil {
+		return nil, nil, err
+	}
+
+	return svc.ds.ListHostPastActivities(ctx, hostID, opt)
```

Comment:
I think that as we did for upcoming activities, we need to clear out the ListOptions so that only supported ones are set (same as https://github.com/fleetdm/fleet/pull/16183/files/f0d383a65ffdf80736981950d81e21ce1f8c2df7..8e47515a6e2feb46eb2f5fe15ea175ca06a42aca#diff-c7892a43478f422de35f07e1e9a45ae123bf461bdbf83f8b560229ed3548a77fR90-R98).

Reply:
Done!

Reply:
I think "past" activities must be newest first, unlike the upcoming activities (upcoming shows the order in which the scripts will run, past shows the most recent activity first): https://www.figma.com/file/Ab3yKMQJMjmVSUWJMwlubP/%2315529-Upcoming-activities%3A-Target-scripts-on-hosts-whether-they're-online-or-offline?type=design&node-id=435-12041&mode=design&t=wyaNi0xtXxZk45UD-0